lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosSobolIndices.cxx

```diff
@@ -65,111 +66,119 @@ inline bool varianceComparison(const std::pair<UnsignedInteger, Scalar> &a, cons
   return a.second > b.second;
 }
 
+/* String formatter with blank space right padding */
+String FunctionalChaosSobolIndices::padString(const String & string) const
+{
+  const String intermediateString = OSS() << string;
+  const UnsignedInteger columnWidth = ResourceMap::GetAsUnsignedInteger("FunctionalChaosSobolIndices-PrintColumnWidth");
+  const UnsignedInteger rightPadding = intermediateString.size() < columnWidth ? columnWidth - intermediateString.size() : 0;
+  OSS oss;
+  oss << intermediateString << String(rightPadding, ' ');
+  return oss;
+}
+
 String FunctionalChaosSobolIndices::__str__(const String & /*offset*/) const
 {
-  const UnsignedInteger inputDimension = functionalChaosResult_.getDistribution().getDimension();
+  return __repr_markdown__();
+}
+
+String FunctionalChaosSobolIndices::__repr_markdown__() const
+{
+  const Distribution inputDistribution = functionalChaosResult_.getDistribution();
+  const UnsignedInteger inputDimension = inputDistribution.getDimension();
+  const Description inputDescription = inputDistribution.getDescription();
   const UnsignedInteger outputDimension = functionalChaosResult_.getMetaModel().getOutputDimension();
   OSS oss;
+  oss << FunctionalChaosSobolIndices::GetClassName() << "\n";
 
   const Indices indices(functionalChaosResult_.getIndices());
   const Sample coefficients(functionalChaosResult_.getCoefficients());
   const UnsignedInteger basisSize = indices.getSize();
+  EnumerateFunction enumerateFunction(functionalChaosResult_.getOrthogonalBasis().getEnumerateFunction());
 
-  // compute the variance
-  Point variance(outputDimension);
-  Point mean(outputDimension);
-  for (UnsignedInteger i = 0; i < outputDimension; ++i)
-  {
-    for (UnsignedInteger k = 0; k < basisSize; ++ k)
-      // Take into account only non-zero indices as the null index is the mean of the vector
-      if (indices[k] > 0)
-        variance[i] += coefficients(k, i) * coefficients(k, i);
-      else
-        mean[i] = coefficients(k, i);
-  }
-
-  // standard deviation
+  // compute the mean, the variance, the standard deviation
+  const FunctionalChaosRandomVector fcRandomVector(functionalChaosResult_);
+  const Point mean = fcRandomVector.getMean();
+  const CovarianceMatrix covarianceMatrix = fcRandomVector.getCovariance();
   Point stdDev(outputDimension);
-  for (UnsignedInteger i = 0; i < outputDimension; ++ i)
-    stdDev[i] = std::sqrt(variance[i]);
+  for (UnsignedInteger i = 0; i < outputDimension; ++i)
+    stdDev[i] = std::sqrt(covarianceMatrix(i, i));
 
   // quick summary
-  oss << " input dimension: " << inputDimension << "\n"
-      << " output dimension: " << outputDimension << "\n"
-      << " basis size: " << functionalChaosResult_.getReducedBasis().getSize() << "\n"
-      << " mean: " << mean.__str__() << "\n"
```

Comment:
No need to call `__str__()` if you instantiate `oss` with the proper flag: `OSS oss(false);`

Reply:
Fixed at 17b243fba